lib/src/git.rs

```diff
@@ -2014,45 +2014,103 @@ fn rename_remote_git_refs(
+
+    let mut new_remote = remote
+        .repo()
+        .remote_at(url)
+        .and_then(|r| {
+            // we either need to set the new push url or copy the old one.
+            let push_url = match push_url {
+                Some(u) => gix::Url::try_from(u).map(Some)?,
+                None => {
+                    // here, we have to decide whether the push url was explicitly set in the old
+                    // remote definition. We do that by comparing the fetch and push urls and just
+                    // assume that if they're the same, push url was not set explicitly.
+                    //
+                    // There's really no other way of doing that using the gix API because it
+                    // returns the fetch url when asked for a push url that is not explicitly set.
+                    let old_push_url = remote.url(gix::remote::Direction::Push);
+                    let old_fetch_url = remote.url(gix::remote::Direction::Fetch);
+                    if old_push_url != old_fetch_url {
+                        old_push_url.map(gix::Url::to_owned)
+                    } else {
+                        None
+                    }
+                }
+            };
+
+            // only set the push url if it is different from the fetch url.
+            // This is to keep the git config idiomatic and, perhaps more importantly, to
+            // not mess with the (arguably more common) git configs that only have the fetch
+            // url set. We have no way of distinguishing a "no push url set"
+            // from "push url is the same as fetch url" using the gix API.
+            //
+            // we know the fetch url is set here, because we passed it to
+            // the remote_at method above.
+            match push_url {
+                Some(push_url) if &push_url != r.url(gix::remote::Direction::Fetch).unwrap() => {
+                    r.push_url(push_url)
+                }
+                _ => Ok(r),
+            }
+        })
+        .map(|r| r.with_fetch_tags(remote.fetch_tags()))?;
+
+    // Copy the existing data from `remote`.
+    for dir in [gix::remote::Direction::Fetch, gix::remote::Direction::Push] {
         new_remote
             .replace_refspecs(
                 remote
-                    .refspecs(direction)
+                    .refspecs(dir)
                     .iter()
                     .map(|refspec| refspec.to_ref().to_bstring()),
-                direction,
+                dir,
             )
             .expect("existing refspecs to be valid");
     }
+
     Ok(new_remote)
 }
 
-pub fn set_remote_url(
+/// sets the new URLs on the remote. If a URL of given kind is not provided, it
+/// is not changed. I.e. it is not possible to remove a fetch/push URL from a
+/// remote using this method.
+pub fn set_remote_urls(
     store: &Store,
     remote_name: &RemoteName,
-    new_remote_url: &str,
+    new_fetch_url: Option<&str>,
+    new_push_url: Option<&str>,
```

Comment:
nit: maybe rename `new_fetch_url` back to `new_url`? I think it's more correct because (fetch)`url` is used for both fetching and pushing by default.